packages/syft/src/syft/types/syft_object.py

```diff
@@ -395,6 +397,13 @@ class SyftObject(SyftBaseObject, SyftObjectRegistry, SyftMigrationRegistry):
 
     # all objects have a UID
     id: UID
+    _created_date: NewDateTime | None = None
+    _updated_date: NewDateTime | None = None
+    _deleted_date: NewDateTime | None = None
+
+    # @property
+    def is_deleted(self) -> bool:
```

Comment:
@koenvanderveen can you make sure that we address the actual permanent deletion of objects seperately. For example if there was some dangerous code or a very large blob we want to remove that should be possible. I figure the version upgrade surgery tools will be the best place to add this kind of behavior that allows for precise manual changes to data.